wfe/wfe.go

```diff
@@ -999,10 +999,10 @@ func (wfe *WebFrontEndImpl) FinalizeOrder(
 		return
 	}
 
-	// The existing order must be in a pending status to finalize it
-	if orderStatus != acme.StatusPending {
+	// The existing order must be in a ready status to finalize it
+	if orderStatus != acme.StatusReady {
 		wfe.sendError(acme.MalformedProblem(fmt.Sprintf(
-			"Order's status (%q) was not pending", orderStatus)), response)
+			"Order's status (%q) was not 'ready'", orderStatus)), response)
 		return
 	}
```

Comment:
I've changed this in-line with the spec:

```
Once all of the
   authorizations listed in the order object are in the "valid" state,
   the order transitions to the "ready" state.  The order moves to the
   "processing" state after the client submits a request to the order's
   "finalize" URL and the CA begins the issuance process for the
   certificate.  Once the certificate is issued, the order enters the
   "valid" state.
```

Let me know if I've misinterpreted anything here!

Reply:
Looks like master doesn't mention 'ready' either: https://github.com/certbot/certbot/blob/master/acme/acme/messages.py#L139-L147

I'm not sure what Boulder does here 😬 I'd hope it supports 'ready'?

Reply:
Ahhh. The 'ready' state was newly added in v10 of the spec...

Previously (in 09):
```
status (required, string):  The status of this order.  Possible
      values are: "pending", "processing", "valid", and "invalid".
```

Pebble's README states it intends to implement an ACME v09 client. Is this current? Or does it aim to track the latest draft of the spec?

What version of the spec does Boulder implement, and what version does Letsencrypt's v2 staging and production endpoint implement? 😬

Reply:
>  Or does it aim to track the latest draft of the spec?

It aims to track the latest.

> What version of the spec does Boulder implement, and what version does Letsencrypt's v2 staging and production endpoint implement?

Basically draft-10 but with a [couple noted divergences](https://community.letsencrypt.org/t/acme-v2-production-environment-wildcards/55578) (See "specification divergences"), one of which is this "Ready" status. I have a PR out on Boulder to fix that one: https://github.com/letsencrypt/boulder/pull/3614